api/src/main/java/org/openmrs/module/registrationcore/api/impl/RegistrationCoreServiceImpl.java

```diff
@@ -488,10 +486,17 @@ private Patient persistImportedMpiPatient(MpiPatient mpiPatient) {
 	private IdentifierSourceService getIssAndUpdateIdSource() {
 		IdentifierSourceService iss = Context.getService(IdentifierSourceService.class);
 		if (idSource == null) {
-			Integer idSourceId = registrationCoreProperties.getOpenMrsIdentifierSourceId();
-			idSource = iss.getIdentifierSource(idSourceId);
+			Object sourceIdentifier = null;
+			try {
+				sourceIdentifier = registrationCoreProperties.getOpenMrsIdentifierSourceId();
+				idSource = iss.getIdentifierSource((Integer) sourceIdentifier);
+			}
+			catch (APIException e) {
+				sourceIdentifier = registrationCoreProperties.getOpenMrsIdentifierSourceUuid();
+				idSource = iss.getIdentifierSourceByUuid((String) sourceIdentifier);
+			}
 			if (idSource == null) {
```

Comment:
If you make the fix I suggest above, you don't need this try/catch here.  Simply get either the uuid or the id, and then return the corresponding source.  Both of those methods should work.